games-strategy/1oom/patches/1oom-1.0.patchset

```diff
@@ -0,0 +1,526 @@
++    AM_CONDITIONAL(COMPILE_UNIX, false)
++    AM_CONDITIONAL(COMPILE_WIN32, false)
++    AM_CONDITIONAL(COMPILE_MSDOS, false)
++    AM_CONDITIONAL(COMPILE_HAIKU, true)
+ else
+     OS_DIR="\$(top_builddir)/src/os/unix"
+     OS_SRC_DIR="\$(top_srcdir)/src/os/unix"
+@@ -558,6 +580,7 @@ else
+     AM_CONDITIONAL(COMPILE_UNIX, true)
+     AM_CONDITIONAL(COMPILE_WIN32, false)
+     AM_CONDITIONAL(COMPILE_MSDOS, false)
++    AM_CONDITIONAL(COMPILE_HAIKU, false)
+ fi
+ OS_LIBS="$OS_DIR/libos.a"
+ OS_INCLUDES="-I$OS_SRC_DIR"
+diff --git a/src/fmt_mus.c b/src/fmt_mus.c
+index 192bc20..4912205 100644
+--- a/src/fmt_mus.c
++++ b/src/fmt_mus.c
+@@ -36,6 +36,8 @@ struct noteoffs_s {
+ };
+ 
+ #define XMID_TICKSPERQ  60
++#define XMID_FREQ 480
++#define XMID_DEFAULT_QNLEN 500000
+ /*#define XMID_USE_BANKS*/
+ 
+ /* -------------------------------------------------------------------------- */
+@@ -118,7 +120,8 @@ static int xmid_convert_evnt(const uint8_t *data_in, uint32_t len_in, const uint
+     struct noteoffs_s s[1];
+     int rc = -1, noteons = 0, looppoint = -1;
+     uint32_t len_out = 0, t_now = 0, delta_time = 0;
+-    unsigned qnlen = 500000;
++    uint64_t delta_c;
++    unsigned qnlen = XMID_DEFAULT_QNLEN;
+     bool end_found = false;
+     *tune_loops = false;
+     memset(s, 0, sizeof(s[0]));
+@@ -137,6 +140,7 @@ static int xmid_convert_evnt(const uint8_t *data_in, uint32_t len_in, const uint
+             case 0x90:
+                 {
+                     uint32_t dt_off;
++                    uint64_t dtoff_c;
+                     uint8_t b;
+                     dt_off = 0;
+                     skip_extra_bytes = 1;
+@@ -146,7 +150,7 @@ static int xmid_convert_evnt(const uint8_t *data_in, uint32_t len_in, const uint
+                         ++skip_extra_bytes;
+                     }
+                     dt_off |= b;
+-                    uint64_t dtoff_c = (uint64_t)dt_off * 480 * 500000 * 2 / ((uint64_t)qnlen * XMID_TICKSPERQ);
++                    dtoff_c = (uint64_t)dt_off * XMID_FREQ * XMID_DEFAULT_QNLEN * 2 / ((uint64_t)qnlen * XMID_TICKSPERQ);
+ 					dt_off = dtoff_c >> 1;
+ 					if (dtoff_c & 0x1) {
+ 						dt_off++;
+@@ -274,7 +278,7 @@ static int xmid_convert_evnt(const uint8_t *data_in, uint32_t len_in, const uint
+                     delta_time += *data_in++;
+                     --len_in;
+                 }
+-                uint64_t delta_c = (uint64_t)delta_time * 480 * 500000 * 2 / ((uint64_t)qnlen * XMID_TICKSPERQ);
++                delta_c = (uint64_t)delta_time * XMID_FREQ * XMID_DEFAULT_QNLEN * 2 / ((uint64_t)qnlen * XMID_TICKSPERQ);
+ 				delta_time = delta_c >> 1;
+ 				if (delta_c & 0x1) {
+ 					delta_time++;
+@@ -285,10 +289,11 @@ static int xmid_convert_evnt(const uint8_t *data_in, uint32_t len_in, const uint
+         if (add_event) {
+             uint32_t len_delta_time;
+             uint8_t buf_delta_time[4];
++            uint32_t delay_noff;
```

Comment:
Such a change looks like it's done to please gcc2, but this actually build with gcc8, doesn't it?

Reply:
No, that change is part of the fix to the XMID->MID conversion so it's compatible with Haiku's native midi implementation.  Haiku seems to have some note off/on timing weirdness that is not present on other platforms.  I can't vouch for whether the whole thing will compile under GCC2 since I don't have the time/resources to test myself but I'm not going to add code to a project that won't compile under GCC2 myself.

Not entirely sure how it is relevant in any case.